time.go

```diff
@@ -101,13 +102,45 @@ func (t UnixTime) String() string {
 // must never happen. The panic is here to prevent from broken setup to be
 // processing data incorrectly.
 func IsExpired(ctx Context, t UnixTime) bool {
-	blockNow, ok := BlockTime(ctx)
-	if !ok {
-		panic("block time is not present")
+	blockNow, err := BlockTime(ctx)
+	if err != nil {
+		panic(fmt.Sprintf("%+v", err))
 	}
 	return t <= AsUnixTime(blockNow)
 }
 
+// InThePast returns true if given time is in the past compared to the current
+// time as declared in the context. Context "now" should come from the block
+// header.
+// Keep in mind that this function is not inclusive of current time. It given
+// time is equal to "now" then this function returns false.
+// This function panic if the block time is not provided in the context. This
+// must never happen. The panic is here to prevent from broken setup to be
+// processing data incorrectly.
+func InThePast(ctx context.Context, t time.Time) bool {
+	now, err := BlockTime(ctx)
+	if err != nil {
+		panic(fmt.Sprintf("%+v", err))
```

Comment:
IMHO we should fail downstream when an empty block time is set in: https://github.com/iov-one/weave/blob/master/app/store.go#L336 
It can not be empty with any standard Tendermint instance. WDYT @husio, @ethanfrey  ?

Reply:
Adding this check to were it is set is a good idea.

It is good to have it here as well in case you forget to set it up for tests. Otherwise, you may spend a lot of time on a failing test without an obvious reason why it fails.

Reply:
See https://github.com/tendermint/tendermint/blob/v0.31.5/consensus/state.go#L1699 , 
https://github.com/tendermint/tendermint/blob/v0.31.5/state/state.go#L146-L150

Reply:
You are write. Note that requiring this when set (good idea) will force us to set it in all abci tests, so may require a bit of test cleanup. As long as it doesn't turn into a huge job, let's do it

Reply:
What is the outcome? Should I add a check to https://github.com/iov-one/weave/blob/master/app/store.go#L336
```go
if req.Header.GetTime().IsZero() { panic("missing block time") } 
```

Reply:
Please add this check.
If this breaks more tests than you care to address in this PR, then remove the check and make an issue to do this soon.

Reply:
Done in https://github.com/iov-one/weave/pull/675/commits/d7c0fda2d5272eebcc774185ec9f4ca524ef0865